### ISSUE TITLE
HTML5 Boilerplate

### ISSUE THREAD

**Issue by blt41** [CS3398-Sapphire-Sourcerer/Group-Project#164]
11/1/2016 ---> 6:13 am to - 6:49 research
Learning about different options in html. Differences of adobe dreamweaver and adobe muse and if they are able to be used with Pycharm. There is a major different in dreamweaver and Muse. Muse if for people that don't really use code or edit the code much but want to create something using predefined templates ( not for programmers) whilst dreamweaver is more for the programmer and can do both ( Muse like characteristic of already predefined templates ) but one can edit and dabble in the code editing much more in this enviroment. Although decided to look more into Pycharm and how to use front end development in that more than these options given that the team had decided to use Pycharm.  So I ran into jetbrains and found some interesting ways of using front end design.
I will be using html and css as my langauges in the front end development:
https://www.jetbrains.com/help/pycharm/2016.1/live-editing-of-html-css-and-javascript.html
This has more about editing of html, css, and javascript in pycharm using templates.
https://www.jetbrains.com/help/pycharm/2016.1/tutorial-creating-and-applying-live-templates-code-snippets.html
Learning about live templates in this section on jetbrains. Why? templates can be very useful later on in my sprint 2 development. They seem to be an important part of pycharm and utilizing it as well in development. 
Learning about html5 the web's most popular front end design. 
HTML5 Boilerplate helps you build fast, robust, and adaptable web apps or sites. Kick-start your project with the combined knowledge and effort of 100s of developers, all in one little package.
★ Analytics, icons, and more

A lean, mobile-friendly HTML template; optimized Google Analytics snippet; placeholder touch-device icon; and docs covering dozens of extra tips and tricks.
★ Normalize.css and helpers

Includes Normalize.css — a modern, HTML5-ready alternative to CSS resets — and further base styles, helpers, media queries, and print styles.
★ jQuery and Modernizr

Get the latest minified versions of two best-of-breed libraries: jQuery (via Google’s CDN, with local fallback) and the Modernizr feature detection library.
★ High performance
This also will work with the github integrations used by the team better, and it will be much cleaner of an option than previous researched options. 
https://html5boilerplate.com/
Research about what an EditorConfig is? Why does it help developers? 
EditorConfig helps developers define and maintain consistent coding systles between editors and IDEs. the editorconfig project consists of a file format for definding coding styles and a collection of text editor plugins that enable editors to read the file format and adhere to defined styles. Editor Config files are easily readable and they work nicely with version control systems.
Apache settings to help you deliver excellent site performance. We independently maintain server configs, and an ant build script. An example of a .editorconfig is the file setting end-of-line and indentation styles for python and javascript files... and so on...
This is needed considering that in our team work we will be dabbling in such a world of many different files whilst dreamweaver and Muse is more static and unable to do so.
I will also be researching the core of HTML5 Boilerplate
Here there is a list of subjects I can study to understand the HTML, CSS Javascript and XML ( an introduction to crossdomain requests). Here I can go into guides to default html, css, and understand what is XML? ( We see it being generated earlier when dabbling with front end html design)? Why is it used?
What is htaacces? all about the apache web server configs.
I will research how to Extend and customize HTML5 Boilderplate. Here is some useful advice on how to make my project with Boilderplate even better. Things such as app stores, dns prefetching, google universal analytics, internet explorer( as we know this is its own universe), news feeds, search engines, social network( integrations), URLS adapting to this enviroment, and web app development.
The ChangeLOG has every change of html 5.0 that has been updated and changed.
Who uses HTML5 Boilerplate?
Google, Microsoft, NASA, Nike, Barack Obama, ITV News, Creative Commons, Australia Post, Racing Green, and many more.
Lol Barack Obama guys... this is important. Needless to say it seems to be an important front-end tool.
http://www.initializr.com/
There is more... Start an HTML5 project in 15 seconds! Apparently.
Initializr is an HTML5 templates generator to help you getting started with a new project based on HTML5 Boilerplate. It generates for you a clean customizable template with just what you need to start
http://getbootstrap.com/getting-started/
Compiling CSS and JavaScript

Bootstrap uses Grunt for its build system, with convenient methods for working with the framework. It's how we compile our code, run tests, and more.
Installing Grunt

To install Grunt, you must first download and install node.js (which includes npm). npm stands for node packaged modules and is a way to manage development dependencies through node.js.
Then, from the command line:

    Install grunt-cli globally with npm install -g grunt-cli.
    Navigate to the root /bootstrap/ directory, then run npm install. npm will look at the package.json file and automatically install the necessary local dependencies listed there.

When completed, you'll be able to run the various Grunt commands provided from the command line.
Available Grunt commands
grunt dist (Just compile CSS and JavaScript)

Regenerates the /dist/ directory with compiled and minified CSS and JavaScript files. As a Bootstrap user, this is normally the command you want.
grunt watch (Watch)

Watches the Less source files and automatically recompiles them to CSS whenever you save a change.
grunt test (Run tests)

Runs JSHint and runs the QUnit tests headlessly in PhantomJS.
grunt docs (Build & test the docs assets)

Builds and tests CSS, JavaScript, and other assets which are used when running the documentation locally via bundle exec jekyll serve.
grunt (Build absolutely everything and run tests)

Compiles and minifies CSS and JavaScript, builds the documentation website, runs the HTML5 validator against the docs, regenerates the Customizer assets, and more. Requires Jekyll. Usually only necessary if you're hacking on Bootstrap itself.
Troubleshooting

Should you encounter problems with installing dependencies or running Grunt commands, first delete the /node_modules/ directory generated by npm. Then, rerun npm install.
Basic template

Start with this basic HTML template, or modify these examples. We hope you'll customize our templates and examples, adapting them to suit your needs.

Copy the HTML below to begin working with a minimal Bootstrap document.
Copy

<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    
    <title>Bootstrap 101 Template</title>

    
    <link href="css/bootstrap.min.css" rel="stylesheet">

    
    
    
  </head>
  <body>
    <h1>Hello, world!</h1>

    
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
    
    <script src="js/bootstrap.min.js"></script>
  </body>
</html>

Examples

Build on the basic template above with Bootstrap's many components. We encourage you to customize and adapt Bootstrap to suit your individual project's needs.

Get the source code for every example below by downloading the Bootstrap repository. Examples can be found in the docs/examples/ directory.
I just downloaded Bootstrap itself from the three different skill levels offered.
http://getbootstrap.com/css/
Some of the ways Bootstrap makes us eof HTML elements and CSS properties in projects.
Global CSS settings, fundamental HTML elements styled and enhanced with extensible classes, and an advanced grid system.
http://getbootstrap.com/examples/jumbotron/#
Templates our team can learn from on how to lay our pages out.
How navigation bars are made with Boot strap
http://getbootstrap.com/components/#navbar
the Framework and using the frame work is intense with templates that explain how to do many different front end options.  Such as customizing a navbar and carousel like feature like the roll over feature I showed you all with the rolloverexample earlier. How to create a sign-inpage using this feature. A cover page ( a one page template for building simple and beautiful home pages. A blog. 
http://getbootstrap.com/customize/?id=d3ae6b797ec30179a29cbb90c9e97106
Lets review how to compile and configure out Bootstrap....
Customize and download
Customize Bootstrap's components, Less variables, and jQuery plugins to get your very own version.

**Comment by blt41** [comment]
This was just the start of my research, and as I find out more about how we can utilize this possibly in front-end design I will add issues to this epic.
